docker/configure_workers_and_start.py

```diff
@@ -490,46 +635,171 @@ def generate_worker_files(
     new_worker_types = []
     for worker_type in worker_types:
         if ":" in worker_type:
-            x = worker_type.split(":")
-            y = 0
-            # should only be 2 components, a type of worker and an integer as a
+            worker_type_components = split_and_strip_string(worker_type, ":")
+            count = 0
+            # Should only be 2 components, a type of worker(s) and an integer as a
             # string. Cast the number as an int then it can be used as a counter.
-            if len(x) == 2 and x[-1].isdigit():
-                y = int(x[1])
+            if (
+                len(worker_type_components) == 2
+                and worker_type_components[-1].isdigit()
+            ):
+                count = int(worker_type_components[1])
             else:
                 error(
                     "Multiplier signal(:) for worker found, but incorrect components: "
                     + worker_type
+                    + ". Please fix."
                 )
             # As long as there are more than 0, we add one to the list to make below.
-            while y > 0:
-                new_worker_types.append(x[0])
-                y -= 1
+            while count > 0:
+                new_worker_types.append(worker_type_components[0])
+                count -= 1
         else:
             # If it's not a real worker, it will error out below
             new_worker_types.append(worker_type)
 
     # worker_types is now an expanded list of worker types.
     worker_types = new_worker_types
+
     # For each worker type specified by the user, create config values
     for worker_type in worker_types:
-        worker_config = WORKERS_CONFIG.get(worker_type)
-        if worker_config:
-            worker_config = worker_config.copy()
+        # pre-template the worker_config so when updating we don't get a KeyError
+        worker_config: Dict[str, Any] = {
+            "app": "",
+            "listener_resources": [],
+            "endpoint_patterns": [],
+            "shared_extra_conf": {},
+            "worker_extra_conf": "",
+        }
+
+        # Peel off any name designated before a '=' to use later.
+        requested_worker_name = ""
+        if "=" in worker_type:
+            # Split on "=", remove extra whitespace from ends then make list
+            worker_type_split = split_and_strip_string(worker_type, "=")
+            if len(worker_type_split) > 2:
+                error(
+                    "To many worker names requested for a single worker, or to many "
+                    "'='. Please fix: " + worker_type
+                )
+            # if there was no name given, this will still be an empty string
+            requested_worker_name = worker_type_split[0]
+            # Uncommon mistake that will cause problems. Name string containing spaces.
+            if len(requested_worker_name.split(" ")) > 1:
```

Comment:
or just `if " " in requested_worker_name:`

Reply:
That is much easier to look at. Fixed.